tuf/repository_lib.py

```diff
@@ -1506,9 +1523,27 @@ def generate_snapshot_metadata(metadata_directory, version, expiration_date,
       # list these roles found in the metadata directory.
       if tuf.roledb.role_exists(rolename, repository_name) and \
           rolename not in ['root', 'snapshot', 'timestamp', 'targets']:
-        fileinfodict[metadata_name] = get_metadata_versioninfo(rolename,
+
+        length = None
```

Comment:
I think we could remove these two assignments and the IF statement altogether.

Reply:
I rebased and addressed that comment.